### PR TITLE
API CHANGE: remove some Camera, Engine, and View deprecated APIs

### PR DESCRIPTION
**Pull request by bejado**

### CODE REVIEW

**Review by romainguy — APPROVED** [review]

**Review by pixelflinger — APPROVED** [review]

**Comment by bejado** [comment]
Updated so the Java API now mirrors the C++ API. Added `Camera.getEntity` Java API.

**Review by prideout — APPROVED** [review]

File: android/filament-android/src/main/cpp/Engine.cpp [hunk 1]
```diff
@@ -143,12 +137,14 @@ Java_com_google_android_filament_Engine_nGetCameraComponent(JNIEnv*, jclass,
...
+extern "C" JNIEXPORT void JNICALL
 Java_com_google_android_filament_Engine_nDestroyCamera(JNIEnv*, jclass,
         jlong nativeEngine, jlong nativeCamera) {
     Engine* engine = (Engine*) nativeEngine;
-    Camera *camera = (Camera *) nativeCamera;
-    engine->destroy(camera);
+    Camera* camera = (Camera*) nativeCamera;
+    Entity c = camera->getEntity();
+    engine->destroyCameraComponent(c);
+    EntityManager::get().destroy(c);
```
[the remark this answers is not in the document]
Comment (bejado) [reply android/filament-android/src/main/cpp/Engine.cpp]: What do you mean? I removed `destroyCamera` from Java, now the only API method is `destroyCameraComponent`, which mirrors the C++ side and only destroys the `Camera`, not the entity.

File: android/filament-utils-android/src/main/java/com/google/android/filament/utils/ModelViewer.kt [hunk 2]
```diff
@@ -105,7 +105,7 @@ class ModelViewer(val engine: Engine) : android.view.View.OnTouchListener {
     init {
         renderer = engine.createRenderer()
         scene = engine.createScene()
-        camera = engine.createCamera().apply { setExposure(kAperture, kShutterSpeed, kSensitivity) }
+        camera = engine.createCamera(EntityManager.get().create()).apply { setExposure(kAperture, kShutterSpeed, kSensitivity) }
```
[the remark this answers is not in the document]
Comment (bejado) [reply android/filament-utils-android/src/main/java/com/google/android/filament/utils/ModelViewer.kt:108]: SG

File: android/filament-android/src/main/cpp/Engine.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (prideout) [reply android/filament-android/src/main/cpp/Engine.cpp]: Argh, this comment was an old draft that I never intended to send (I started a gChat discussion instead).  When I approved the commit GitHub published the comment.

[the remark this answers is not in the document]
Comment (bejado) [reply android/filament-android/src/main/cpp/Engine.cpp]: All good :)

File: android/filament-utils-android/src/main/java/com/google/android/filament/utils/ModelViewer.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (bejado) [reply android/filament-utils-android/src/main/java/com/google/android/filament/utils/ModelViewer.kt:108]: I just realized that Engine.java doesn't have a `getEntityManager`, but probably a good time to add it

**Pull request merged**